web-app/src/test/com.isa.aem/calculatorMethod/CreateAListOfAvailableCurrenciesTest.java

```
@@ -0,0 +1,117 @@
+import com.isa.aem.FileContentReader;
+import com.isa.aem.LoadCurrencyNameCountryFlags;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.jupiter.api.DisplayName;
+
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;;
+
+public class CreateAListOfAvailableCurrenciesTest {
+    private CurrencyRepository currencyRepository = new CurrencyRepository();
+    private FileContentReader fileContentReader = new FileContentReader();
+    private CreateAListOfAvailableCurrencies createAListOfAvailableCurrencies;
+
+    @Before
+    public void init() {
+        fileContentReader.readFile();
+        fileContentReader.addPLNToListCurrency();
+        LoadCurrencyNameCountryFlags loadCurrencyNameCountryFlags = new LoadCurrencyNameCountryFlags();
+    }
+
+    @Test
+    @DisplayName("Should return list on size more then zero when currency is correct")
+
+    public void returnsListWithSizeMorThenZeroIfCurrencyExist() {
+        createAListOfAvailableCurrencies = new CreateAListOfAvailableCurrencies();
+        // arrange
+        String currency = "PLN";
+        // act
+        Boolean result = createAListOfAvailableCurrencies.availableCurrencyObjects(currency).size() > 0;
+        // assert
+        assertEquals(true, result);
+    }
+
+    @Test
+    @DisplayName("Should return list on size more then zero when currency is correct and write in lower case")
+
+    public void returnsListWithSizeMorThenZeroIfCurrencyExistAndWhenIsWriteInLowerCase() {
+        createAListOfAvailableCurrencies = new CreateAListOfAvailableCurrencies();
+        // arrange
+        String currency = "pln";
+        // act
+        Boolean result = createAListOfAvailableCurrencies.availableCurrencyObjects(currency).size() > 0;
+        // assert
+        assertEquals(true, result);
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    @DisplayName("Should return throw when currency have not correct name")
```

napisalas, powinien zwrocic rzucic kiedy cos tam...sprobuj jedno z ponizszych (pierwsza wersja chyba lepsza tym bardziej, ze nic nie zwracasz):
- should throw an exception jakis tam kiedy cos tam
- should return an exception jakis tam kiedy cos tam (czyli zwrocic rzucony wyjatek)

---

poprawione